### PR TITLE
fishPlugins.wakatime-fish: init at 0.0.3

### PR DESCRIPTION
**Pull request by ocfox**
###### Description of changes
https://github.com/ik11235/wakatime.fish

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/shells/fish/plugins/wakatime-fish.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+  src = fetchFromGitHub {
+    owner = "ik11235";
+    repo = "wakatime.fish";
+    rev = "v${version}";
+    hash = "sha256-t0b8jvkNU7agF0A8YkwQ57qGGqcYJF7l9eNr12j2ZQ0=";
+  };
+
+  buildInputs = [
+    wakatime
```
Comment (natsukium) [inline pkgs/shells/fish/plugins/wakatime-fish.nix:15]: Looking at other plugins (including zsh), it seems that we should not add the external command to `buildInputs`.

**Review by natsukium — APPROVED** [review]
It works fine with `wrapFish`.

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 224606` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>fishPlugins.wakatime-fish</li>
  </ul>
</details>

File: pkgs/shells/fish/plugins/wakatime-fish.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  src = fetchFromGitHub {
+    owner = "ik11235";
+    repo = "wakatime.fish";
+    rev = "v${version}";
+    hash = "sha256-t0b8jvkNU7agF0A8YkwQ57qGGqcYJF7l9eNr12j2ZQ0=";
+  };
+
+  preFixup = ''
+    substituteInPlace $out/share/fish/vendor_conf.d/wakatime.fish \
+      --replace "if type -p wakatime" "if type -p ${lib.getExe wakatime}" \
```
Comment (SuperSandro2000) [inline pkgs/shells/fish/plugins/wakatime-fish.nix:20]: We could patch out the entire detection.

**Pull request merged**